### ISSUE TITLE
Support Docker Compose V2

### ISSUE THREAD

**Issue by StefanFl** [DefectDojo/django-DefectDojo#4862]
**Bug description**
With Docker Compose V2 (see https://docs.docker.com/compose/cli-command/), the commands `build` and `up` don't work anymore but are showing error messages.

**Steps to reproduce**

1. Enable Docker Compose V2 in Docker for Windows
2. Run `docker-compose build` or `docker-compose up`
3. See error

**Expected behavior**
Build or run the docker images

**Deployment method** *(select with an `X`)*
- [x] Docker
- [ ] Kubernetes
- [ ] GoDojo
- [ ] setup.bash / legacy-setup.bash

**Environment information**
 - Operating System: Microsoft Windows 10
 - DefectDojo version 2.0.3

**Console logs**

```
WARN[0000] The NGINX_VERSION variable is not set. Defaulting to a blank string.
WARN[0000] The NGINX_METRICS_ENABLED variable is not set. Defaulting to a blank string.
WARN[0000] The DD_PORT variable is not set. Defaulting to a blank string.
WARN[0000] The DD_TLS_PORT variable is not set. Defaulting to a blank string.
WARN[0000] The DJANGO_VERSION variable is not set. Defaulting to a blank string.
WARN[0000] The DD_DJANGO_METRICS_ENABLED variable is not set. Defaulting to a blank string.
WARN[0000] The DD_ALLOWED_HOSTS variable is not set. Defaulting to a blank string.
WARN[0000] The DD_DATABASE_URL variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CELERY_BROKER_USER variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CELERY_BROKER_USER variable is not set. Defaulting to a blank string.
WARN[0000] The DD_SECRET_KEY variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CREDENTIAL_AES_256_KEY variable is not set. Defaulting to a blank string.
WARN[0000] The DD_MEDIA_ROOT variable is not set. Defaulting to a blank string.
WARN[0000] The DJANGO_VERSION variable is not set. Defaulting to a blank string.
WARN[0000] The DD_DATABASE_URL variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CELERY_BROKER_USER variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CELERY_BROKER_USER variable is not set. Defaulting to a blank string.
WARN[0000] The DD_SECRET_KEY variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CREDENTIAL_AES_256_KEY variable is not set. Defaulting to a blank string.
WARN[0000] The DJANGO_VERSION variable is not set. Defaulting to a blank string.
WARN[0000] The DD_DATABASE_URL variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CELERY_BROKER_USER variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CELERY_BROKER_USER variable is not set. Defaulting to a blank string.
WARN[0000] The DD_SECRET_KEY variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CREDENTIAL_AES_256_KEY variable is not set. Defaulting to a blank string.
WARN[0000] The DD_MEDIA_ROOT variable is not set. Defaulting to a blank string.
WARN[0000] The DJANGO_VERSION variable is not set. Defaulting to a blank string.
WARN[0000] The DD_DATABASE_URL variable is not set. Defaulting to a blank string.
WARN[0000] The DD_ADMIN_USER variable is not set. Defaulting to a blank string.
WARN[0000] The DD_ADMIN_USER variable is not set. Defaulting to a blank string.
WARN[0000] The DD_ADMIN_FIRST_NAME variable is not set. Defaulting to a blank string.
WARN[0000] The DD_ADMIN_LAST_NAME variable is not set. Defaulting to a blank string.
WARN[0000] The DD_INITIALIZE variable is not set. Defaulting to a blank string.
WARN[0000] The DD_SECRET_KEY variable is not set. Defaulting to a blank string.
WARN[0000] The DD_CREDENTIAL_AES_256_KEY variable is not set. Defaulting to a blank string.
WARN[0000] The DD_DATABASE_URL variable is not set. Defaulting to a blank string.
WARN[0000] The DD_DATABASE_USER variable is not set. Defaulting to a blank string.
WARN[0000] The DD_DATABASE_PASSWORD variable is not set. Defaulting to a blank string.
WARN[0000] The DD_DATABASE_NAME variable is not set. Defaulting to a blank string.
yaml: line 42: did not find expected alphabetic or numeric character
```

**Comment by StefanFl** [comment]
Docker compose V2 is available as a release candidate. There are currently 2 issues:
- Unit tests are not working because of https://github.com/docker/compose-cli/issues/1994  
- Some environment variables need to be quoted

**Comment by damiencarol** [comment]
@StefanFl do we have PR that cover this one?

**Comment by valentijnscholten** [comment]
Fixed in dev / 2.3.0
